services/worker-manager/test/provider_azure_test.js

```diff
@@ -657,13 +686,22 @@ helper.secrets.mockSuite(testing.suiteName(), [], function(mock, skipping) {
 
   suite('removeWorker', function() {
     let worker, ipName, nicName, vmName;
+    const sandbox = sinon.createSandbox({});
+
     setup('create un-provisioned worker', async function() {
       const workerPool = await makeWorkerPool();
       const workerInfo = {
         existingCapacity: 0,
         requestedCapacity: 0,
       };
+
+      // prevent the worker from being immediately provisioned
+      sandbox.stub(provider, 'checkWorker').returns('ok');
```

Comment:
this just ensures that internal state for fake calls are not interfered by calling `checkWorker` inside `provision()` call